tests/report_tests/plugin_tests/openstack/openstack.py

```diff
@@ -0,0 +1,145 @@
+#
+# This copyrighted material is made available to anyone wishing to use,
+# modify, copy, or redistribute it subject to the terms and conditions of
+# version 2 of the GNU General Public License.
+#
+# See the LICENSE file in the source distribution for further information.
+import json
+
+from sos_tests import StageTwoReportTest, ubuntu_only
+from sos.utilities import shell_out
+
+from avocado.utils import distro
+from avocado.utils.software_manager import distro_packages
+
+
+class OpenstackConfScrubbedTest(StageTwoReportTest):
+    """Ensure that the various openstack confs are picked up and properly
+    scrubbed
+
+    :avocado: tags=stagetwo
+    """
+
+    sos_cmd = ('-o openstack_cinder,openstack_keystone,openstack_glance,'
+               'openstack_neutron,openstack_nova')
+    files = [
+        ('cinder.conf', '/etc/cinder/cinder.conf'),
+        ('keystone.domain1.conf',
+         '/etc/keystone/domains/keystone.domain1.conf'),
+        ('keystone.conf', '/etc/keystone/keystone.conf'),
+        ('glance-api.conf', '/etc/glance/glance-api.conf'),
+        ('neutron.conf', '/etc/neutron/neutron.conf'),
+        ('nova.conf', '/etc/nova/nova.conf'),
+    ]
+
+    packages = {
+        'rhel': ['openstack-cinder', 'openstack-glance',
+                 'openstack-heat-common', 'openstack-keystone',
+                 'openstack-nova'],
+        'Ubuntu': ['cinder-common', 'glance-common', 'heat-common',
+                   'keystone-common', 'nova-common']
+    }
+
+    openstack_only = True
+
+    def setup_mocked_packages(self):
+        this_distro = distro.detect()
+        if this_distro.name == "centos" or this_distro.name == "centos-stream":
+            shell_out("dnf config-manager --enable crb")
+            shell_out("dnf config-manager --enable powertools")
+            pkg_to_install = "centos-release-openstack-yoga"
+            installed = distro_packages.install_distro_packages(
+                {this_distro.name: [pkg_to_install]})
+            if not installed:
+                raise Exception(
+                    "Unable to install requested packages "
+                    f"{pkg_to_install}")
+            self._write_file_to_tmpdir(
+                'mocked_osp_package', f'["{pkg_to_install}"]')
+        super().setup_mocked_packages()
+
+    def teardown_mocked_packages(self):
+        this_distro = distro.detect()
+        if this_distro.name == "centos" or this_distro.name == "centos-stream":
+            pkgs = self.read_file_from_tmpdir('mocked_osp_package')
+            if not pkgs:
+                return
+            pkgs = json.loads(pkgs)
+            for pkg in pkgs:
+                self.sm.remove(pkg)
+        super().teardown_mocked_packages()
+
+    def test_cinder_conf_collected_and_scrubbed(self):
+        self.assertFileCollected('/etc/cinder/cinder.conf')
+
+        keys_to_mask = [
+            'cmB4zBYq3VWFMNqNKFLcqS5Zq8ystLLsTd5BFLbCtX67qShnhgHFxxRFjkhbY54x',
```

Comment:
I think this is how we are doing other tests with masks as well. We may need to have an overhaul of the tests to do it key based instead of value based. It easier to search for the value instead of the key.